src/lib/routes/ready-check.ts

```diff
@@ -0,0 +1,62 @@
+import type { Request, Response } from 'express';
+import type { IUnleashConfig } from '../types/option.js';
+import type { IUnleashServices } from '../services/index.js';
+import type { Db } from '../db/db.js';
+import type { Logger } from '../logger.js';
+
+import Controller from './controller.js';
+import { NONE } from '../types/permissions.js';
+import { createResponseSchema } from '../openapi/util/create-response-schema.js';
+import type { ReadyCheckSchema } from '../openapi/spec/ready-check-schema.js';
+
+export class ReadyCheckController extends Controller {
+    private logger: Logger;
+
+    private db?: Db;
+
+    constructor(
+        config: IUnleashConfig,
+        { openApiService }: Pick<IUnleashServices, 'openApiService'>,
+        db?: Db,
+    ) {
+        super(config);
+        this.logger = config.getLogger('ready-check.js');
+        this.db = db;
+
+        this.route({
+            method: 'get',
+            path: '',
+            handler: this.getReady,
+            permission: NONE,
+            middleware: [
+                openApiService.validPath({
+                    tags: ['Operational'],
+                    operationId: 'getReady',
+                    summary: 'Get instance readiness status',
+                    description:
+                        'This operation returns information about whether this Unleash instance is ready to serve requests or not. Typically used by your deployment orchestrator (e.g. Kubernetes, Docker Swarm, Mesos, et al.).',
+                    responses: {
+                        200: createResponseSchema('readyCheckSchema'),
+                        500: createResponseSchema('readyCheckSchema'),
```

Comment:
503 maybe? Feels more semantically correct but won't split hairs if you like 500